http2-client.cabal

```diff
@@ -27,7 +27,7 @@ library
                      , bytestring >= 0.10 && < 1
                      , containers >= 0.5 && < 1
                      , deepseq >= 1.4 && < 2
-                     , http2 >= 1.6 && < 2.1
+                     , http2 >= 1.6 && < 3.1
```

Comment:
There will be module namespace conflicts here. The correct way to support GHC-9 and http2-3.x is to rework the module naming scheme for this library, I think. While I'm aware that package namespacing exists, requiring users of this library to use that in order to use this library may be more onerous than necessary. This is especially true since some of the functionality is being replicated by both libraries.

Reply:
Yes, I agree that using `PackageImports` may be the bigger evil here.